### ISSUE TITLE
Product ID of Game

### ISSUE THREAD

**Issue by marcreichel** [xboxapi/Python-Wrapper#23]
Hi there,

great API so far :)

BUT:
how can I find out the product_id of a game?
Is it possible to search for a game?

I need the product_id to fetch further information about the game.

Thanks for any help.

**Comment by ghost** [comment]
Are you looking for game details for a game owned by a user or just in general? If you're looking for game details by `XUID` (if the player owns the game) you can call `get('xboxonegames')` or `get('xbox360games')` depending on what you're looking for and parse that game out. 

I don't believe a method exists yet for searching directly for Xbox Game titles by themselves. @djekl does an endpoint exist for doing game lookups by the title?

**Comment by marcreichel** [comment]
Thanks for the reply.

I need to search for a game a user potentially do not own.
Also `get('xboxonegames')` and `get('xbox360games')` do not include the `Product_ID` so I can not make a request to get the game description and images etc.

And another problem with `get('xboxonegames')` and `get('xbox360games')` is that these endpoints only contain games where the user has gamerscore. Games with 0 gamerscore are not included in the resultset. Could this be changed?
I want to use these lists of games to verify a user can create an event for a specific game on our website. But at the moment the user can only do this for games where he has gamerscore.

**Comment by marcreichel** [comment]
I would also appriciate it to lookup game details with the `titleID` of a game :)

**Comment by ReddHerring** [comment]
Hi,

you can lookup game details by converting the titleId to hex via.:
hexvalue = hex(titleId).split('x')[-1]
Then you can use the link for "game-details-hex/[hexvalue]"

But about the first problem. I am also looking for a solution to get all owned/bought games. If you buy a game at the store, but you don't install and play it. You can't get the information that you own this game.

Is there a possibility to get all bought items/games?

By the way! Thank you for this API generally. I'ts great, and I like it :)

**Comment by marcreichel** [comment]
> hexvalue = hex(titleId).split('x')[-1]

Great tip 👍 Thanks :) I will give this a try :)

**Comment by marcreichel** [comment]
Whoops. Closed it by mistake :D

**Comment by ghost** [comment]
I am unable to make modifications to the backend API, I only maintain this Python library for accessing the API. Although, I just added the `game-details*` endpoints to this library they were missing. You can try them out by pulling down branch `feature/game-details`. I will wait on merging these changes to master until you give them a test.

At this point I think it is best to move this issue over to https://github.com/xboxapi/Xbox-One-API-Issues/issues since a backend API change is needed to display the additional info. Once the backend change has been made this Python library should just work, in the case it doesn't we'll want to create a new issue here.

Thanks again for showing interest in this library 👍 

/cc @djekl

**Comment by marcreichel** [comment]
I'm not a Python developer :D I just didn't know where to open an issue :D 
Thanks for your help. I opened an Issue in the other repository :)

**Comment by ghost** [comment]
No problem at all. I am glad I could be of some help anyways.

**Comment by ReddHerring** [comment]
Hey, I come back to the asked question from @marcreichel  in this issue, "how to get the titleId of a game?" above.

The calls `get('xboxonegames')` or `get('xbox360games')` obtain all ever started games with your profile, with one different I detected. The call for the 360 games include games you havn't earned achievements. The call for the XBox One games doesn't support this. There are only your games with earned achievements.

So a good way to get detailed informations for all your games is to call the `titlehub-achievement-list`, supported from the api. There you get your whole achievement list. And a game will be added to this list, when you start it for the first time. So you could write a python function to read all titleIds from this list. After that send a request for the game-details to the api.

I think, that's the best way to get an overwiev over all games you have, at the moment. Perhaps there will be an update at the api, to get ONE games with a gamerscore of 0 from the call `get('xboxonegames')`??
